src/main/java/ch/wisv/areafiftylan/products/controller/TicketRestController.java

```diff
@@ -119,13 +119,35 @@ public TicketRestController(TicketService ticketService, OrderService orderServi
     @PreAuthorize("hasRole('ADMIN')")
     @PostMapping("/types")
     public ResponseEntity<?> addTicketType(@RequestBody @Validated TicketType type) {
-
         TicketType ticketType = ticketService.addTicketType(type);
-
         return createResponseEntity(HttpStatus.CREATED, "TicketType successfully added", ticketType);
     }
 
     @PreAuthorize("hasRole('ADMIN')")
+    @GetMapping("/types")
+    public Collection<TicketType> readTicketTypes() {
+        return ticketService.getAllTicketTypes();
+    }
+
+    @PreAuthorize("hasRole('ADMIN')")
+    @PutMapping("/types/{typeId}")
+    public ResponseEntity<?> updateTicketType(@PathVariable Long typeId, @RequestBody @Validated TicketType type) {
+        TicketType ticketType = ticketService.updateTicketType(typeId, type);
+        if (ticketType == null) {
```

Comment:
The service returns null if the save was not done successfully (when typeId doesn't exist for example)